### ISSUE TITLE
Send an object to the server

### ISSUE THREAD

**Issue by pat2381** [Job79/EasyTcp#73]
Hello, I'm really great about what you did here. It's really easy to work with. 
But I have one question: The client sends an action like this: 

`client.SendAction ("USER", "Hello everyone");` 

I would like to send a class object to the server, e.g. 

`client.SendAction ("USER", MyCLass);`

and the method like this;

```
[EasyAction("USER")] // Trigger function when "USER" action is received
public void EchoAction(Message message)
{
    MyClase user = message as MyClass
    user.name 
}
```

is that possible?

**Comment by Job79** [comment]
Good evening,

This should be possible using the `client.SendAction(string, object)` and `message.To<type>()` functions.
Here is an example:
```
char[] items = { 't', 'e', 's', 't', ' ', 'a', 'r', 'r', 'a', 'y' };
client.Send(items);

server.OnDataReceive += (_, m) => {
    char[] arr = m.To<char[]>()
};
```

Depending on the .net version used you might need to set the `Serialize` and `Deserialize` options of the server / client.
In .net core 3.1 and .net 5 JSON is used by default as serialization method.

**Comment by pat2381** [comment]
Hi there, 
thank you for your reply. If it really works that easily, that would be really great. 
I will try this!

**Comment by pat2381** [comment]
Hi there,
I've tried it now. It doesn't work directly with an Object I have to Serialize it first. So far all good!
But I noticed something else. The client event "OnConnect" doesn't seem to fire in connection with an EasyTcpActionServer

```
private void ConnectToServer()
{
       client = new EasyTcpClient();
       if (!client.Connect("127.0.0.1", 7530)){ return;};

         client.OnDisconnect += Client_OnDisconnect;
         client.OnConnect += Client_OnConnect;
 }

private void Client_OnConnect(object sender, EasyTcpClient e)
{
      SetConsoleText($"Connect to Server...{e.GetEndPoint()}", Status.INFO);
}

```
What I also noticed is that if I have an EasyTcpActionServer and I want to respond to an action, nothing is sent to the client.

```
[EasyAction("User")] // Trigger function when "ECHO" action is received
public void EchoAction(object sender, EasyTcp4.Message message)
{
     SetConsoleText($"Client has send", Status.Info);
     message.Client.SendAction("RETURN", "Toll");
 }
```

Does the server also need a client to send? Can you please explain this to me!

**Comment by Job79** [comment]
The messages should be send to the client, can't reproduce the issue. Could you send a complete server / client?

The serialization probably didn't work because the package didn't detect .net 6 which has json serialization. This should be fixed by commit 12d0851b66cfe5e7a4b6d193035628c1ba56e008
